File: backend/src/main/java/io/chesslave/mouth/WebSpeechSynthesis.java
package io.chesslave.mouth;


import io.chesslave.app.Event;
import io.chesslave.app.EventBus;

public class WebSpeechSynthesis implements SpeechSynthesis {

    private final EventBus events;

    public WebSpeechSynthesis(EventBus events) {
        this.events = events;
    }

    @Override
    public void speak(Utterance utterance) {
        events.fire(Event.of("speak", utterance));
    }
}

File: backend/src/main/java/io/chesslave/mouth/WebSpeechSynthesis.kt
package io.chesslave.mouth

import io.vertx.core.json.JsonObject
import io.vertx.reactivex.core.eventbus.EventBus

class WebSpeechSynthesis(
    private val events: EventBus
) : SpeechSynthesis {

    override fun speak(utterance: Utterance) {
        events.publish("speak", JsonObject.mapFrom(utterance))
    }
}
